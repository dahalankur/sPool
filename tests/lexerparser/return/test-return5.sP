# returning a lambda within a lambda
(string -> (int -> int)) stoi = lambda (int -> int) (string intStr):
    return lambda int (): 
        return 3;;
