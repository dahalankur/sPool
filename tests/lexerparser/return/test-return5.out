PROGRAM {[ 
    DEFINE(ARROWTY(STRINGTY -> ARROWTY(INTTY -> INTTY)), stoi, LAMBDA(ARROWTY(INTTY -> INTTY), FORMALS((STRINGTY, intStr)), [ 
        RETURN(LAMBDA(INTTY, FORMALS(), [ 
            RETURN(INT(3))]))]))]}
